src/Extractor/ResponseExtractor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\JsonRpc\Extractor;
+
+final class ResponseExtractor extends Extractor
+{
+    public function getErrorCode() : ?int
+    {
+        $error = $this->messageContents['error'] ?? null;
```

Comment:
všechny tyhle řádky lze zahodit a zkrátit to na `return $this->messageContents['error']['code'] ?? null;`